tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virtual/VirtualThreadServlet.java

```diff
@@ -0,0 +1,493 @@
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(annoThread), "Should be impossible to get a virtual thread on Java 17");
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(ddThread), "Should be impossible to get a virtual thread on Java 17");
+            return;
+        }
+
+        // Java 21+
+        assumingThat(isVirtual(ddThread), () -> {
+            
+            // Test schedule on virtual threads
+            final LinkedBlockingQueue<Object> results = new LinkedBlockingQueue<>();
+            ScheduledFuture<?> oneTimeFuture = virtualManagedScheduledExecutorDD.schedule(
+                    (Runnable) new LookupActionCaptureThread(results,
+                            "java:app/concurrent/ManagedScheduledExecutorDDVirtual"),
+                    TestConstants.pollInterval.toMillis(), TimeUnit.MILLISECONDS);
+    
+            assertTimeoutPreemptively(TestConstants.waitTimeout, () -> {
+                for (; 2 != results.size(); Wait.sleep(TestConstants.pollInterval)) {
+                    // Wait till we have exactly 2 results indicating 1 run of schedule
+                }
+            });
+    
+            assertTrue(oneTimeFuture.isDone());
+            assertEquals(2, results.size());
+    
+            Thread thread = (Thread) results.poll(1, TimeUnit.MILLISECONDS);
+            Object result = results.poll(1, TimeUnit.MILLISECONDS);
+    
+            if (result instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result);
+            
+            // Avoid assertion that a task was executed on a virtual thread since there is no guarantee
+            if (isVirtual((Thread) thread))
+                log.info("ManagedScheduledExecutorService.schedule() resulted in task being run on a virtual thread.");
+
+        });
+        
+        assumingThat(isVirtual(annoThread), () -> {
+            // Test scheduleAtFixedRate on virtual threads
+            final LinkedBlockingQueue<Object> resultsFixedRate = new LinkedBlockingQueue<>();
+            ScheduledFuture<?> future = virtualManagedScheduledExecutorAnno.scheduleAtFixedRate(
+                    new LookupActionCaptureThread(resultsFixedRate, "java:app/concurrent/ManagedScheduledExecutorAnnoVirtual"),
+                    0, TestConstants.pollInterval.toMillis(), TimeUnit.MILLISECONDS);
+    
+            assertTimeoutPreemptively(TestConstants.waitTimeout, () -> {
+                for (; 6 >= resultsFixedRate.size(); Wait.sleep(TestConstants.pollInterval)) {
+                    // Wait till we have at least 6 results indicating 3 runs of schedule
+                }
+            });
+    
+            Wait.waitCancelFuture(future); // Cancel execution of timer
+    
+            int resultsAfterCancel = resultsFixedRate.size();
+            assertTrue(resultsAfterCancel >= 6, "Should have executed schedule at least 3 times.");
```

Comment:
```suggestion
            assertTrue(resultsAfterCancel >= 6, "Should have executed schedule at least 6 times.");
```

Reply:
There are two results added for each execution.  So this is correct.